libraries/AP_RangeFinder/AP_RangeFinder_MSP.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_MSP.h"
+#include <AP_HAL/AP_HAL.h>
+
+#if  HAL_MSP_ENABLED
+
+extern const AP_HAL::HAL& hal;
+
+/*
+   The constructor also initialises the rangefinder. Note that this
+   constructor is not called until detect() returns true, so we
+   already know that we should setup the rangefinder
+*/
+AP_RangeFinder_MSP::AP_RangeFinder_MSP(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    AP_RangeFinder_Backend(_state, _params)
+{
+    state.last_reading_ms = AP_HAL::millis();
```

Comment:
I wonder why we need to set this last reading in the constructor..

Reply:
sorry for the (very) late reply @rmackay9 , to be honest I copied this code from the mavlink rangefinder backend.
In my personal experience I usually initialize a time reference in the constructor to delay triggering an expiration event which would be otherwise triggered immediately for (now - time_referecene) > delay would always be true when time_reference == 0
I don't know if in this specific case it's of any importance